src/hotspot/share/runtime/vmOperations.cpp

```diff
@@ -463,6 +453,10 @@ void VM_Exit::doit() {
 
   set_vm_exited();
 
+  // The ObjectMonitor subsystem uses perf counters so do this before
+  // we call exit_globals() so we don't run afoul of perfMemory_exit().
+  ObjectSynchronizer::do_final_audit_and_print_stats();
```

Comment:
Given this used to be done in the prolog, is there anything above this line that might interfere with it somehow? Or more simply put, why not do this first so that it is as close as possible to where it used to be?

Reply:
Thanks. I missed the fact that the old code was only an async deflation request.